### DISCUSSION TITLE
Carbon lexer for CudaText

### DISCUSSION THREAD

**Discussion by Alexey-T** [carbon-language/carbon-lang#1607] [General]
It is just info. I made the Carbon lexer for CudaText (free editor - https://cudatext.github.io/ ). It is in the Addon Manager (Plugins / Addons Manager / Install).

Lexer was tested on bunch of files from Carbon repo, I guessed many keywords from context, I maybe missed some keywords.
Lexer shows some items in the code-tree: class, function, interface, impl (IMO showing the items 'namespace' will be not nice).

![carb](https://user-images.githubusercontent.com/4154978/180599659-757819f7-a8a2-4385-a0dd-7c6aeb75a7a0.png)

Where can I find full list of language keywords ? Standard types? Standard modifiers (public private inline…)?

**Comment by jonmeow** [comment]
https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/lexical_conventions/words.md is intended to be a complete list. Note some likely keywords may not be in the list because they haven't been added to the design yet. `public` though may not end up being a keyword, instead being the default.

**Comment by Alexey-T** [comment]
Updating keyword list. now words
```
api
const
explicit
extern
external
inline
private
public
final
abstract
override
partial
```
have another style, like 'modifiers', not like 'usual keywords'.

**Comment by Alexey-T** [comment]
Fixed rules in lexer for

- numbers (hex, binary, float too)
- raw-strings with `#` chars
